### PR TITLE
Logging *up* to specified level (and not just at specified level).

### PR DESCRIPTION
**Pull request by icrc-fdeniger**
`initializer.logging.*` properties are expected in the code

See more details on Slack here: https://openmrs.slack.com/archives/CPC20CBFH/p1672332612265309

### CODE REVIEW

**Comment by icrc-fdeniger** [comment]
For modification in [InitializerLogConfigurator2_4.java](https://github.com/mekomsolutions/openmrs-module-initializer/pull/215/files#diff-09be3da9372d42a91e8707552b36dcf4b3972513fec8f3bf0eed029b4e3e5c94)
The filter used was a Level Filter: only log with the selected level will be accepted.
It should be replaced by a LevelRangeFilter to filter log level lower than the selected level.

**Comment by icrc-fdeniger** [comment]
For UnitTest: they are not compatible with Windows OS. I fixed only one test

File: api/src/main/java/org/openmrs/module/initializer/api/loaders/BaseInputStreamLoader.java [hunk 1]
```diff
@@ -34,6 +34,7 @@ protected File getLoadedFile() {
 	
 	@Override
 	protected void load(File file) throws Exception {
+		log.info("Load file {}", file.getAbsolutePath());
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/module/initializer/api/loaders/BaseInputStreamLoader.java:37]: `log.info("Loading file {}", file.getAbsolutePath());`

("loading" not "load")

Comment (icrc-fdeniger) [reply api/src/main/java/org/openmrs/module/initializer/api/loaders/BaseInputStreamLoader.java]: fixed

File: api-2.4/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_4.java [hunk 2]
```diff
@@ -42,7 +43,7 @@ private Appender getFileAppender(Level level, Path logFilePath) {
 		
 		Appender appender = FileAppender.newBuilder().setName(logFilePath.getFileName().toString())
 		        .withFileName(logFilePath.toString()).setLayout(layout)
-		        .setFilter(LevelMatchFilter.newBuilder().setLevel(level).build()).build();
+		        .setFilter(LevelRangeFilter.createFilter(null, level, Filter.Result.ACCEPT, null)).build();
```
Comment (icrc-fdeniger) [inline api-2.4/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_4.java:46]: here we can replace the first parameter with Level.FATAL to be sure to have all logs

**Review by mks-d — APPROVED** [review]
Leaving the last word to @ibacher. Thanks so much @icrc-fdeniger!

File: api-2.4/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_4.java (hunk 2 above)
Comment (ibacher) [reply api-2.4/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_4.java:46]: I think if you mean that this should be `LevelRangeFilter.createFilter(level, Level.FATAL, ...)`, I think that matches the original intention, i.e., that the level is the lowest level specified.

Comment (icrc-fdeniger) [reply api-2.4/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_4.java:46]: I believe it should be `.setFilter(LevelRangeFilter.createFilter(Level.FATAL, level,..` as Level.FATAL is lower one

Comment (icrc-fdeniger) [reply api-2.4/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_4.java:46]: https://logging.apache.org/log4j/2.x/log4j-api/apidocs/src-html/org/apache/logging/log4j/spi/StandardLevel.html#line.29

File: api/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_0.java [hunk 3]
```diff
@@ -46,12 +46,12 @@ private Appender getFileAppender(Level level, Path logFilePath) {
 			
 			// since the org.apache.log4j.varia package doesn't exist in the log4j2 bridge, we need to use reflection
 			// so this class only has a runtime dependency on the LevelMatchFilter
-			Filter levelMatchFilter = (Filter) Class.forName("org.apache.log4j.varia.LevelMatchFilter").getConstructor()
+			Filter levelRangeFilter = (Filter) Class.forName("org.apache.log4j.varia.LevelRangeFilter").getConstructor()
```
Comment (icrc-fdeniger) [inline api/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_0.java:49]: no yet fully tested. On Going...

Comment (icrc-fdeniger) [reply api/src/main/java/org/openmrs/module/initializer/api/logging/InitializerLogConfigurator2_0.java:49]: can't test it as I can't use initializer 2.4 with openmrs core 2.3.3

```
 org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'conversionService': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: java.util.Set org.openmrs.module.uiframework.UiFrameworkConversionServiceFactoryBean.springManagedConverters; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.openmrs.ui.framework.converter.PatientToSimpleObjectConverter#0': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.openmrs.ui.framework.UiUtils org.openmrs.ui.framework.converter.PatientToSimpleObjectConverter.ui; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'uiUtils': Injection of autowired dependencies failed; nested exception is org.springframework.beans.factory.BeanCreationException: Could not autowire field: org.springframework.context.MessageSource org.openmrs.ui.framework.BasicUiUtils.messageSource; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'initializer.InitializerMessageSource': Invocation of init method failed; nested exception is java.lang.LinkageError: loader constraint violation: loader (instance of org/openmrs/module/ModuleClassLoader) previously initiated loading for a different type with name "org/apache/commons/io/filefilter/IOFileFilter"
```

**Review by ibacher — APPROVED** [review]

**Pull request merged**